### PR TITLE
thread SecretStore into SDS

### PR DESCRIPTION
**Pull request by quanjielin**
https://github.com/istio/istio/issues/5903

### CODE REVIEW

File: security/pkg/nodeagent/cache/secretcache.go [hunk 1]
```diff
@@ -27,6 +27,11 @@ import (
 // The size of a private key for a leaf certificate.
 const keySize = 2048
 
+// SecretStore defines interface that secret storages need to implement.
```
Comment (incfly) [inline security/pkg/nodeagent/cache/secretcache.go]: Could you rephrase it into what SecretStores provides more specifically? The interface should states the user expected functionality, and disregard the implementor.

File: security/pkg/nodeagent/cache/secretcache.go [hunk 2]
```diff
@@ -27,6 +27,11 @@ import (
...
+type SecretStore interface {
+	GetSecret(proxyID, token string) (*SecretItem, error)
```
Comment (incfly) [inline security/pkg/nodeagent/cache/secretcache.go]: From the name indicates, the `SecretStore` is supposed to have both `Get` and `Put` methods. Reading the code, I understand it's because we send CSR in case of cache miss. And the SecretStore implementation itself have a map store, https://github.com/istio/istio/blob/collab-gcp-identity/security/pkg/nodeagent/cache/secretcache.go#L92

So it feels more like a `SecretManager`? So that the caching optimization is not visible to the users.

File: security/pkg/nodeagent/cache/secretcache.go (hunk 1 above)
Comment (quanjielin) [reply security/pkg/nodeagent/cache/secretcache.go]: renamed and updated.

File: security/pkg/nodeagent/cache/secretcache.go (hunk 2 above)
Comment (quanjielin) [reply security/pkg/nodeagent/cache/secretcache.go]: thanks, renamed.
